File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/configuration/rabbitmq/docs/exchange/AnimalListener.kt
package io.micronaut.configuration.rabbitmq.docs.exchange

// tag::imports[]
import io.micronaut.configuration.rabbitmq.annotation.Queue
import io.micronaut.configuration.rabbitmq.annotation.RabbitListener
import io.micronaut.context.annotation.Requires
import java.util.*

// end::imports[]

@Requires(property = "spec.name", value = "CustomExchangeSpec")
// tag::class[]
@RabbitListener // <1>
class AnimalListener {

    internal var receivedAnimals: MutableList<Animal> = Collections.synchronizedList(ArrayList())

    @Queue("cats") // <2>
    fun receive(cat: Cat) { // <3>
        receivedAnimals.add(cat)
    }

    @Queue("snakes") // <2>
    fun receive(snake: Snake) { // <3>
        receivedAnimals.add(snake)
    }
}
// end::class[]

File: docs-examples/example-python/src/test/python/micronaut/rabbitmq/docs/exchange/AnimalListener.py
from micronaut.context.annotation import Requires
# tag::imports[]
from micronaut.rabbitmq.annotation import Queue, RabbitListener

from .Animal import Animal
from .Cat import Cat
from .Snake import Snake
# end::imports[]


@Requires(property="spec.name", value="CustomExchangeSpec")
# tag::clazz[]
@RabbitListener  # <1>
class AnimalListener:

    def __init__(self) -> None:
        self.received_animals: list[Animal] = []

    @Queue("cats")  # <2>
    def receive_cat(self, cat: Cat) -> None:  # <3>
        self.received_animals.append(cat)

    @Queue("snakes")  # <2>
    def receive_snake(self, snake: Snake) -> None:  # <3>
        self.received_animals.append(snake)
# end::clazz[]
